server/src/omtDocumentInformationProvider.ts

```diff
@@ -0,0 +1,359 @@
+        const characterIndex = line.search(new RegExp(`${declaredObject}(?=[^a-zA-Z0-9]|$)`)) ?? line.indexOf(declaredObject);
+
+        if (characterIndex >= 0) {
+            documentLinks.push({
+                name: declaredObject,
+                range: Range.create({ line: l, character: characterIndex }, { line: l, character: characterIndex + declaredObject.length })
+            });
+        }
+    });
+    return documentLinks;
+}
+
+export function getAvailableObjectsFromDocument(document: TextDocument, shorthands?: Map<string, string>): OmtAvailableObjects {
+    const documentText = document.getText();
+    const yamlDocument = parse(document.getText());
+    const availableImports: OmtImport[] = [];
+    const definedObjects: OmtLocalObject[] = [];
+    if ("import" in yamlDocument && shorthands) {
+        const documentImports = yamlDocument["import"];
+        const importUrls = Object.keys(documentImports);
+        importUrls.forEach(importUrl => {
+            const imports: string[] = documentImports[importUrl];
+            const uriMatchResult = getUriMatch(' ' + importUrl, document, shorthands);
+            if (uriMatchResult) {
+                imports.forEach(importName => {
+                    availableImports.push({ name: importName, url: importUrl, fullUrl: uriMatchResult.url });
+                });
+            }
+        });
+    }
+    if ("queries" in yamlDocument) {
+        definedObjects.push(...findDefinedObjects(yamlDocument["queries"], documentText, "QUERY"));
+    }
+    if ("commands" in yamlDocument) {
+        definedObjects.push(...findDefinedObjects(yamlDocument["commands"], documentText, "COMMAND"));
+    }
+    if ("model" in yamlDocument) {
+        const modelEntries = yamlDocument["model"];
+        definedObjects.push(...findModelEntries(modelEntries, documentText));
+        const keys = Object.keys(modelEntries);
+        keys.forEach(key => {
+            const entry = modelEntries[key];
+
+            if ("commands" in entry) {
+                definedObjects.push(...findDefinedObjects(entry["commands"], documentText, "COMMAND"));
+            }
+
+            if ("queries" in entry) {
+                definedObjects.push(...findDefinedObjects(entry["queries"], documentText, "QUERY"));
+            }
+        });
+    }
+    return { availableImports, definedObjects };
+}
+
+function findModelEntries(modelEntries: any, documentText: string): OmtLocalObject[] {
+    const localDefinedObjects: OmtLocalObject[] = [];
+    const keys = Object.keys(modelEntries);
+    keys.forEach(key => {
+        const range = findRangeWithRegex(documentText, new RegExp(`(${key})(?=: !)`));
+        localDefinedObjects.push({ name: key, range });
+    });
+    return localDefinedObjects;
+}
+
+function findDefinedObjects(value: string, documentText: string, define: string): OmtLocalObject[] {
+    const localDefinedObjects: OmtLocalObject[] = [];
+    const queriesString: any = value;
```

Comment:
Why do you assign the value parameter to a constant here, and why is it of type any, when the parameter is a string?

Reply:
It was not necessary to have this construction, I will remove it :+1: